### PR TITLE
Replace std::cout for errors with messaget::error()

### PR DESCRIPTION
**Pull request by smowton**
Also replace assert(false) with throw while I'm there.

### CODE REVIEW

File: src/solvers/flattening/arrays.cpp [hunk 1]
```diff
@@ -535,8 +535,8 @@ void arrayst::add_array_constraints_with(
 
     if(index_expr.type()!=value.type())
     {
-      std::cout << expr.pretty() << '\n';
-      assert(false);
+      prop.error() << expr.pretty() << messaget::eom;
```
Comment (thk123) [inline src/solvers/flattening/arrays.cpp]: @rjmunro Could this be why the pretty was dumped in the middle of the json (or is this old news?)

**Comment by thk123** [comment]
I see this is to TGS. It is currently difficult to update the submodule pointer since diffblue/test-gen#525 is needed to get Kurts changes in and this is blocked as tests are failing. Depending on exactly how urgent this is we may need to have a think exactly how we get this into a test-gen release.

File: src/solvers/flattening/arrays.cpp (hunk 1 above)
Comment (smowton) [inline src/solvers/flattening/arrays.cpp]: Yes that's the purpose

**Comment by martin-cs** [comment]
This is suitable for a hot-fix.  For an actual fix it would be good to replace the asserts / throw with INVARIANTs (specifically DATA_INVARIANTs by the look of it).

**Comment by martin-cs** [comment]
Approved once the checks are sorted.

**Comment by rjmunro** [comment]
@martin-cs: The plan is to merge this, then follow up with another PR to replace the asserts.

**Comment by martin-cs** [comment]
@smowton explained; hence the structure of my response.

**Pull request merged**